control-plane-operator/controllers/hostedcontrolplane/cvo/reconcile.go

```diff
@@ -478,3 +490,35 @@ func ReconcileServiceMonitor(sm *prometheusoperatorv1.ServiceMonitor, ownerRef c
 
 	return nil
 }
+
+func ReconcileRole(role *rbacv1.Role, ownerRef config.OwnerRef) error {
+	role.Rules = []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{"metrics.k8s.io"},
+			Resources: []string{"pods"},
+			Verbs:     []string{"get"},
+		},
+	}
+	return nil
+}
+
+func ReconcileRoleBinding(rb *rbacv1.RoleBinding, role *rbacv1.Role, ownerRef config.OwnerRef, namespace string) error {
+	rb.RoleRef = rbacv1.RoleRef{
+		APIGroup: rbacv1.SchemeGroupVersion.Group,
+		Kind:     "Role",
+		Name:     role.Name,
+	}
+	rb.Subjects = []rbacv1.Subject{
+		{
+			Kind:      "ServiceAccount",
+			Name:      manifests.ClusterVersionOperatorServiceAccount("").Name,
+			Namespace: namespace,
+		},
+	}
+	return nil
+}
+
+func ReconcileServiceAccount(sa *corev1.ServiceAccount, ownerRef config.OwnerRef) error {
+	ownerRef.ApplyTo(sa)
```

Comment:
this service account also needs a pull secret, see
https://github.com/Davoska/hypershift/blob/0097f51dd8dd3557db6b66008a6ce285dce9a76d/control-plane-operator/controllers/hostedcontrolplane/autoscaler/reconcile.go#L264